src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -889,6 +1041,46 @@ private static Expression BindParameterFromBindAsync(ParameterInfo parameter, Fa
             return Expression.Convert(boundValueExpr, parameter.ParameterType);
         }
 
+        private static Expression BindParameterFromFormFiles(
+            ParameterInfo parameter,
+            FactoryContext factoryContext)
+        {
+            factoryContext.FormRequestBodyParameter = parameter;
+            factoryContext.TrackedParameters.Add(parameter.Name!, RequestDelegateFactoryConstants.FormFileParameter);
+
+            // Do not duplicate the metadata if there are multiple form parameters
+            if (!factoryContext.ReadForm)
+            {
+                factoryContext.Metadata.Add(new AcceptsMetadata(parameter.ParameterType, factoryContext.AllowEmptyRequestBody, FormFileContentType));
+            }
+
+            factoryContext.ReadForm = true;
+
+            return BindParameterFromExpression(parameter, FormFilesExpr, factoryContext, "body");
+        }
+
+        private static Expression BindParameterFromFormFile(
+            ParameterInfo parameter,
+            string key,
+            FactoryContext factoryContext,
+            string trackedParameterSource)
+        {
+            factoryContext.FormRequestBodyParameter = parameter;
```

Comment:
What if there are multiple IFormFile parameters? Does this just clobber the FormRequestBodyParameter?

Reply:
This _should_ handle multiple parameters (there's a test or two for that), but will double-check when I rebase this.

Reply:
You're right - I've refactored this to prefix with `First` and then stopped it being clobbered. It's only used to report a parameter name in a log message if reading the form fails.